datalad_container/containers_run.py

```diff
@@ -112,6 +124,9 @@ def __call__(cmd, container_name=None, dataset=None,
         # with amend inputs to also include the container image
         inputs = (inputs or []) + [image_path]
 
+        old_envvar = os.environ.get(CONTAINER_NAME_ENVVAR, None)
+        os.environ[CONTAINER_NAME_ENVVAR] = container['name']
```

Comment:
What about using patch.dict instead?  Right now the value is cleaned up only if something was already set and wouldn't be restored properly for python callers if there is an exception raised with `Run.__call__()`.

Reply:
Thank you @kyleam for review/suggestions. I am not sure why I was lazy to not use `patch.dict`. Did it (with delayed import, takes too long to import, at least while I am also Zoom'ing). Committed via `--amend`, pushed.